### PR TITLE
for approve 

### PR DESCRIPTION
**Pull request by EuhenySiniy**

### CODE REVIEW

File: src/main/java/controller/DeleteController.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Objects;
+
+@WebServlet("/delete")
+public class DeleteController extends HttpServlet {
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+        String userId = request.getParameter("id");
```
Comment (Taniann) [inline src/main/java/controller/DeleteController.java]: try parse it to Integer with using Optional

File: src/main/java/controller/DeleteController.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        if (Objects.equals(userId, null)) {
```
Comment (Taniann) [inline src/main/java/controller/DeleteController.java]: why not just userId == null

File: src/main/java/controller/DeleteController.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+            request.getRequestDispatcher("/list")
+                    .forward(request, response);
+        } else {
+            int id = Integer.parseInt(userId);
+            UserDao dao = UserDaoImpl.getInstance();
```
Comment (Taniann) [inline src/main/java/controller/DeleteController.java]: better move UserDao to class field

File: src/main/java/controller/UpdateController.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Objects;
+
+@WebServlet("/update")
+public class UpdateController extends HttpServlet {
```
Comment (Taniann) [inline src/main/java/controller/UpdateController.java]: the same problems as in DeleteController

File: src/main/java/dao/UserDao.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import java.util.List;
+
+public interface UserDao {
+    void saveUser(User user);
+
+    void updateUser(User user);
+
+    void deleteUser(int id);
+
+    User findUserById(int id);
```
Comment (Taniann) [inline src/main/java/dao/UserDao.java]: better return `Optional<User>`

File: src/main/java/controller/DeleteController.java (hunk 1 above)
Comment (EuhenySiniy) [reply src/main/java/controller/DeleteController.java]: you mean Optional.of ? Should you send example or info about this, if it possible..)

**Pull request merged**